llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -2286,7 +2295,9 @@ SDValue DAGCombiner::visitPTRADD(SDNode *N) {
         (DAG.isConstantIntBuildVectorOrConstantInt(Y) && N0.hasOneUse()) ||
         (N0.hasOneUse() && Z.hasOneUse() &&
          !DAG.isConstantIntBuildVectorOrConstantInt(Z))) {
-      return DAG.getPointerAdd(DL, X, Add);
+      return Reassociated;
+    } else {
+      DAG.DeleteNode(Reassociated.getNode());
```

Comment:
I think you want deleteAndRecombine so anything created by visiting the ADD also gets recursively deleted